lib/api/core/models/repositories/repository.js

```diff
@@ -1,344 +1,358 @@
+          if (this.databaseEngine === null) {
+            return null;
+          }
+
+          return this.loadOneFromDatabase(id)
+            .then(objectFromDatabase => {
+              if (objectFromDatabase !== null) {
+                this.persistToCache(objectFromDatabase);
+              }
+              return objectFromDatabase;
+            });
+        }
+        this.refreshCacheTTL(object);
+        return object;
+      });
   }
-  if (this.databaseEngine) {
-    promises.push(this.deleteFromDatabase(id));
+
+  /**
+   * Persists the given object in the collection that is attached to the repository.
+   *
+   * @param {Profile|Role|User} object The object to persist
+   * @param {object} opts The persistence options
+   * @returns {Promise}
+   */
+  persistToDatabase (object, opts) {
+    const
+      options = opts || {},
+      method = options.method || 'createOrReplace';
+
+    return this.databaseEngine[method](this.collection, object._id, this.serializeToDatabase(object));
   }
 
-  return Promise.all(promises);
-};
+  /**
+   * Given an object id, delete it from the configured storage engines
+   *
+   * @param {string} id   The id of the object to delete
+   * The opts optional parameters currently accepts only 1 option:
+   *   key: if provided, removes the given key instead of the default one (<collection>/<id>)
+   * @param {object} [opts] optional options for the current operation
+   * @returns {Promise}
+   */
+  delete (id, opts = {}) {
+    const promises = [];
+
+    if (this.cacheEngine) {
+      promises.push(this.deleteFromCache(id, opts));
+    }
+    if (this.databaseEngine) {
+      promises.push(this.deleteFromDatabase(id));
+    }
 
+    return Promise.all(promises);
+  }
 
-/**
- * Delete repository from database according to its id
- * @param id
- */
-Repository.prototype.deleteFromDatabase = function repositoryDeleteFromDatabase (id) {
-  return this.databaseEngine.delete(this.collection, id);
-};
 
-/**
- * Persists the given ObjectConstructor object in cache.
- * The opts optional parameters currently accept 2 options:
- *   key: if provided, stores the object to the given key instead of the default one (<collection>/<id>)
- *   ttl: if provided, overrides the default ttl set on the repository for the current operation.
- *
- * @param {object} object the object to persist
- * @param {object} [opts] optional options for the current operation
- * @returns {Promise}
- */
-Repository.prototype.persistToCache = function repositoryPersistToCache (object, opts) {
-  var
-    options = opts || {},
-    key = options.key || this.cacheKeyPrefix + object._id,
-    ttl = this.ttl;
-
-  if (options.ttl !== undefined) {
-    ttl = options.ttl;
-  }
-  if (ttl === false) {
-    return this.cacheEngine.set(key, JSON.stringify(this.serializeToCache(object)))
-      .then(() => object);
+  /**
+   * Delete repository from database according to its id
+   * @param id
+   */
+  deleteFromDatabase (id) {
```

Comment:
nitpicking: why not adding an optional parameter `collection` here, as you did for `loadOneFromDatabase` method?